app/controllers/api/results_controller.rb

```diff
@@ -6,6 +6,77 @@ def results_search_conditions
       MiqReportResult.for_user(User.current_user).where_clause.ast
     end
 
+    def apply_limit_and_offset(results, options)
+      results.slice(options['offset'].to_i, options['limit'].to_i) || []
+    end
+
+    def sort_params_from(report_result)
+      params['sort_order'] ||= :ascending
```

Comment:
is `params` an external variable?

are you doing it this way to ensure the side effect of modifying the `params`?

If not, could you just use `sort_order` local variable?